simexpal/build.py

```diff
@@ -65,6 +67,15 @@ class Phase(IntEnum):
 	COMPILE = 4
 	INSTALL = 5
 
+def mark_as_finished(basedir, buildname, type = None):
+	if type == None:
+		raise Exception("Wrong key, see util.CONSTANTS for valid keys")
```

Comment:
So this exception will be raised whenever `mark_as_finished()` is called with `type = None` correct? That however is a logical programming error and not something a user would get any helpful information from whenever that would happen. I would recommend changing that to an assertion and **not** raising an exception.